File: populatingNextRightPointersInEachNode.swift
/**
 * Definition for a Node.
 * public class Node {
 *     public var val: Int
 *     public var left: Node?
 *     public var right: Node?
 *	   public var next: Node?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *         self.next = nil
 *     }
 * }
 */

class Solution {
    func connect(_ root: Node?) -> Node? {
        var tempArr:[[Node]] = []
        self.travel(root, tempArr: &tempArr, level: 1)
        return root
    }
    func travel(_ root: Node?, tempArr: inout [[Node]], level: Int) {
        guard root != nil else {
            return
        }
        if tempArr.count < level {
            tempArr.append([])
        }
        root!.next = nil
        tempArr[level-1].last?.next = root
        tempArr[level-1].append(root!)
        travel(root?.left, tempArr: &tempArr, level: level+1)
        travel(root?.right, tempArr: &tempArr, level: level+1)
    }
}

File: LeeCode/populatingNextRightPointersInEachNode.py

# Definition for a Node.
class Node:
    def __init__(self, val, left, right, next):
        self.val = val
        self.left = left
        self.right = right
        self.next = next


class Solution:
    def connect(self, root: 'Node') -> 'Node':
        self.cacheL = 0
        self.cacheArr = []
        self.travle(root, 0)
        for arr in self.cacheArr:
            l = len(arr)
            for i in range(l):
                if i < l-1:
                    arr[i].next = arr[i+1]
                else:
                    arr[i].next = None
        return root

    def travle(self, root: 'Node', level: int):
        if root == None:
            return
        tempArr = None
        if self.cacheL == level:
            tempArr = []
            self.cacheArr.append(tempArr)
            self.cacheL += 1
        else:
            tempArr = self.cacheArr[level]
        tempArr.append(root)
        self.travle(root.left, level+1)
        self.travle(root.right, level+1)
